tinycrypt/ecc_dh.c

```diff
@@ -142,26 +147,27 @@ int uECC_make_key(uint8_t *public_key, uint8_t *private_key)
 
 			/* Converting buffers to correct bit order: */
 			uECC_vli_nativeToBytes(private_key,
-						   BITS_TO_BYTES(NUM_ECC_BITS),
-						   _private);
+							BITS_TO_BYTES(NUM_ECC_BITS),
+							_private);
 			uECC_vli_nativeToBytes(public_key,
-						   NUM_ECC_BYTES,
-						   _public);
+							NUM_ECC_BYTES,
+							_public);
 			uECC_vli_nativeToBytes(public_key + NUM_ECC_BYTES,
- 						   NUM_ECC_BYTES,
-						   _public + NUM_ECC_WORDS);
+ 							NUM_ECC_BYTES,
+							_public + NUM_ECC_WORDS);
 
 			/* erasing temporary buffer that stored secret: */
 			mbedtls_platform_memset(_private, 0, NUM_ECC_BYTES);
 
-			if (private_key == private_key_dup && public_key == public_key_dup) {
+			if (private_key == private_key_dup && public_key == public_key_dup &&
+					rng_num_generated == 2 * NUM_ECC_WORDS*uECC_WORD_SIZE) {
 				return UECC_SUCCESS;
 			}
 			/* Erase key in case of FI */
```

Comment:
Not related to this fixes, but an interesting question:  why do we clean only public key ?

Reply:
The private key is external, and the public key is generated here, so the assumption is that this function only cleans up itself.

Reply:
It seems both are generated in the function https://github.com/PelionIoT/pmi-arm-firmware/blob/55a29859116c52eb6b2a67cea30b2661e82fe69b/libs/mbed-os/features/mbedtls/inc/tinycrypt/ecc_dh.h#L90